shared-module/is31fl3741/IS31FL3741.c

```diff
@@ -248,92 +82,76 @@ void common_hal_displayio_is31fl3741_begin_transaction(is31fl3741_IS31FL3741_obj
 
-// The following are routines to manipulate the IS31FL3741 chip
-// They are not meant to be called by user code but only used
-// internally.
-
-uint8_t cur_page = 99; // set to invalid page to start
-
-void is31fl3741_send_unlock(busio_i2c_obj_t *i2c, uint8_t addr) {
+void common_hal_is31fl3741_send_unlock(is31fl3741_IS31FL3741_obj_t *self) {
     uint8_t unlock[2] = { 0xFE, 0xC5 }; // unlock command
-    common_hal_busio_i2c_write(i2c, addr, unlock, 2);
+    common_hal_busio_i2c_write(self->i2c, self->device_address, unlock, 2);
 }
 
-void is31fl3741_set_page(busio_i2c_obj_t *i2c, uint8_t addr, uint8_t p) {
-    if (p == cur_page) {
+void common_hal_is31fl3741_set_page(is31fl3741_IS31FL3741_obj_t *self, uint8_t p) {
+    if (p == is31fl3741_cur_page) {
         return;
     }
 
-    cur_page = p;
-    is31fl3741_send_unlock(i2c, addr);
+    is31fl3741_cur_page = p;
+    common_hal_is31fl3741_send_unlock(self);
 
     uint8_t page[2] = { 0xFD, 0x00 }; // page command
     page[1] = p;
-    common_hal_busio_i2c_write(i2c, addr, page, 2);
+    common_hal_busio_i2c_write(self->i2c, self->device_address, page, 2);
 }
 
-void is31fl3741_send_enable(busio_i2c_obj_t *i2c, uint8_t addr) {
-    is31fl3741_set_page(i2c, addr, 4);
+void common_hal_is31fl3741_send_enable(is31fl3741_IS31FL3741_obj_t *self) {
+    common_hal_is31fl3741_set_page(self, 4);
     uint8_t enable[2] = { 0x00, 0x01 }; // enable command
-    common_hal_busio_i2c_write(i2c, addr, enable, 2);
+    common_hal_busio_i2c_write(self->i2c, self->device_address, enable, 2);
 }
 
-void is31fl3741_send_reset(busio_i2c_obj_t *i2c, uint8_t addr) {
-    is31fl3741_set_page(i2c, addr, 4);
+void common_hal_is31fl3741_send_reset(is31fl3741_IS31FL3741_obj_t *self) {
+    common_hal_is31fl3741_set_page(self, 4);
     uint8_t rst[2] = { 0x3F, 0xAE }; // reset command
-    common_hal_busio_i2c_write(i2c, addr, rst, 2);
+    common_hal_busio_i2c_write(self->i2c, self->device_address, rst, 2);
 }
 
-void is31fl3741_set_current(busio_i2c_obj_t *i2c, uint8_t addr, uint8_t current) {
-    is31fl3741_set_page(i2c, addr, 4);
+void common_hal_is31fl3741_set_current(is31fl3741_IS31FL3741_obj_t *self, uint8_t current) {
+    common_hal_is31fl3741_set_page(self, 4);
     uint8_t gcur[2] = { 0x01, 0x00 }; // global current command
     gcur[1] = current;
-    common_hal_busio_i2c_write(i2c, addr, gcur, 2);
+    common_hal_busio_i2c_write(self->i2c, self->device_address, gcur, 2);
 }
 
-uint8_t is31fl3741_get_current(busio_i2c_obj_t *i2c, uint8_t addr) {
-    is31fl3741_set_page(i2c, addr, 4);
+uint8_t common_hal_is31fl3741_get_current(is31fl3741_IS31FL3741_obj_t *self) {
+    common_hal_is31fl3741_set_page(self, 4);
     uint8_t gcur = 0x01; // global current command
+    common_hal_busio_i2c_write(self->i2c, self->device_address, &gcur, 1);
+
     uint8_t data = 0;
-    common_hal_busio_i2c_write_read(i2c, addr, &gcur, 1, &data, 1);
+    common_hal_busio_i2c_read(self->i2c, self->device_address, &data, 1);
```

Comment:
It looks like this should be a `write_read`.

Reply:
and remove line 131, I think.

Reply:
The latest build should have removed both and replace them with the `write_read`